distributed/core.py

```diff
@@ -280,7 +280,7 @@ def __await__(self):
         async def _():
             timeout = getattr(self, "death_timeout", 0)
             async with self._startup_lock:
-                if self.status in Status.ANY_RUNNING:
+                if self.status not in {Status.undefined, Status.created, Status.init}:
```

Comment:
I'd rather have a new set in the Status enum, e.g. ``self.status not in Status.NEVER_STARTED``

Reply:
as I mentioned above, this is not the final fix but rather an "FYI something here is broken". imo, the problem is a bit larger than just this particular set